src/utilities/getCurrentDate.ts

```diff
@@ -69,17 +73,35 @@ export function generateDateOrig(currentDate: DateData): Date {
 }
 
 export function generateWeekList(currentDate: DateData): WeekDay[] {
```

Comment:
Got rid of generateWeek() function because we can generate a week with just this one. Also had to get rid of recursion because the device kept running into stack frame error.